src/map.cpp

```diff
@@ -411,6 +421,37 @@ void VoxelStack::Load(Loader &ldr)
 			delete[] this->voxels;
 			this->voxels = (height > 0) ? MakeNewVoxels(height) : nullptr;
 			for (uint i = 0; i < height; i++) this->voxels[i].Load(ldr, version);
+
+			/* In version 3 of VSTK, the fences of the lowest corner of steep slopes have moved from the top voxel to the base voxel. */
+			if (version < 3) {
+
+				static const uint16 low_fences_mask[4] = { // Mask for getting the low fences.
+					(0xf << (4 * EDGE_SE)) | (0xf << (4 * EDGE_SW)), // ISL_TOP_STEEP_NORTH
+					(0xf << (4 * EDGE_SW)) | (0xf << (4 * EDGE_NW)), // ISL_TOP_STEEP_EAST
+					(0xf << (4 * EDGE_NW)) | (0xf << (4 * EDGE_NE)), // ISL_TOP_STEEP_SOUTH
+					(0xf << (4 * EDGE_NE)) | (0xf << (4 * EDGE_SE)), // ISL_TOP_STEEP_WEST
+				};
+
+				for (uint i = 0; i < height; i++) {
+					if (this->voxels[i].GetGroundType() == GTP_INVALID) continue;
+					if (!IsImplodedSteepSlopeTop(this->voxels[i].GetGroundSlope())) continue;
+					uint16 mask = low_fences_mask[this->voxels[i].GetGroundSlope() - ISL_TOP_STEEP_NORTH];
+
+					/* Take out the fences of the top voxel that should be in the base voxel.
+					 * Make the low fences in the high voxel invalid. */
+					uint16 fences = this->voxels[i].GetFences();
+					uint16 lower_fences = fences & mask;
+					uint16 high_invalid = ALL_INVALID_FENCES & mask;
+					mask ^= 0xffff;
```

Comment:
I *could* be exceptionally picky and say that it would be more consistent to use uppercase letters for the hex (as is usual for the rest of the program/prq) but I don't actually care that much :p 
However, I am vaguely concerned about the amount of code that's necessary here for loading "old fences". Are we that concerned about savegame compatibility at this point? If we are, these Load functions are quickly going to become very large, which scares me slightly